src/node_crypto.cc

```diff
@@ -652,14 +652,17 @@ int SSL_CTX_use_certificate_chain(SSL_CTX* ctx,
   if (!x)
     return 0;
 
-  // TODO(addaleax): Turn this into smart pointer as well.
   X509* extra = nullptr;
   int ret = 0;
+  OnScopeLeave cleanup([&extra, &ret] {
+    if (ret)
```

Comment:
There is a potential footgun here.  Someone someday might look at line 686 and see this:
```c++
ret = SSL_CTX_use_certificate_chain(...);
return ret;
```
And change that to:
```c++
return SSL_CTX_use_certificate_chain(...);
```
Without realizing that the assignment to `ret` is necessary because of its side effects.  Probably also easy to miss in code review.

Reply:
I think we could really do a variant here that uses a smart pointer and releases it if necessary; it’s just not trivial to translate the code to that…